File: app/models/projections/mongo/board.rb
module Projections
  module Mongo
    class Board
      include Mongoid::Document

      embedded_in :game, class_name: 'Projections::Mongo::Game'

      embeds_many :tiles, class_name: 'Projections::Mongo::Tile'
      embeds_many :spawns, class_name: 'Projections::Mongo::Spawn'
      embeds_many :checkpoints, class_name: 'Projections::Mongo::Checkpoint'
      embeds_many :robots, class_name: 'Projections::Mongo::Robot'
    end
  end
end

File: app/assets/javascripts/models/board.js
RoboRacer.Models.Board = Backbone.Model.extend({
  initialize: function() {
    this.set('tiles', new RoboRacer.Collections.Tiles());
    this.set('spawns', new RoboRacer.Collections.Spawns());
    this.set('checkpoints', new RoboRacer.Collections.Checkpoints());
    this.set('robots', new RoboRacer.Collections.Robots());
  }
});
